packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -225,6 +225,10 @@ const fetchAggregationData = (page: number): ThunkAction<
         });
       }
     } catch (e) {
+      if ((e as MongoServerError).codeName === 'MaxTimeMSExpired') {
+        track('Aggregation Timed Out');
```

Comment:
Is there any useful metadata to pass here? Maybe the `maxTimeMS` that's set? Fine as is too, mostly raising to see if there's any insights y'all might think of from extra metadata.
If a lot of folks are hitting this event I think that would be useful to see as then maybe we'll want to surface the max time ms setting better (or help people optimize their indexes/pipeline xD).

Reply:
Is there a reason we only want to track the `MaxTimeMSExpired` error? Would tracking all errors with the code name as meta data be an option?

Reply:
Is there a reason we only want to track the `MaxTimeMSExpired` error? Would tracking all errors with the code name as meta data be an option?

Reply:
My understanding is that we don't want to submit just anything to telemetry and want to be more intentional about what gets there. The rationale from Max for adding this one is exactly as you put it: if wee see too many of those, it might indicate that we need to make timeout setting more obvious. My feeling is that not many other errors are like that. I will add a maxtimems value, it definitely can help to see if a lot of people are running with a default timeout